projects/max14906/src/examples/iio_example/iio_example.c

```diff
@@ -0,0 +1,110 @@
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ *  - Neither the name of Analog Devices, Inc. nor the names of its
+ *    contributors may be used to endorse or promote products derived
+ *    from this software without specific prior written permission.
+ *  - The use of this software may or may not infringe the patent rights
+ *    of one or more patent holders.  This license does not release you
+ *    from the requirement that you obtain separate licenses from these
+ *    patent holders to use this software.
+ *  - Use of the software either in source or binary form, must be run
+ *    on or directly connected to an Analog Devices Inc. component.
+ *
+ * THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT,
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+#include "iio_example.h"
+#include "iio_max14906.h"
+#include "common_data.h"
+#include "no_os_print_log.h"
+#include "iio_app.h"
+
+int iio_example_main()
+{
+	int ret;
+
+	/** IIO descriptor and initialization parameter. */
+	struct max14906_iio_desc *max14906_iio_desc;
+
+	struct max14906_iio_desc_init_param max14906_iio_ip = {
+		.max14906_init_param = &max14906_ip,
+		/** Channel configuration to be used in the example. */
+		.channel_configs = {
+			{
+				.enabled = true,
+				.function = MAX14906_OUT,
+			},
+			{
+				.enabled = true,
+				.function = MAX14906_IN,
+			},
+			{
+				.enabled = true,
+				.function = MAX14906_OUT,
+			},
+			{
+				.enabled = true,
+				.function = MAX14906_IN,
+			},
+		},
+	};
+
+	/** IIO app. */
+	struct iio_app_desc *app;
+	struct iio_app_init_param app_init_param = { 0 };
+
+	ret = max14906_iio_init(&max14906_iio_desc, &max14906_iio_ip);
+	if(ret)
+		goto error;
+
+	/** Declaring iio_devices structure */
+	struct iio_app_device iio_devices[] = {
+		{
+			.name = "max14906",
+			.dev = max14906_iio_desc,
+			.dev_descriptor = max14906_iio_desc->iio_dev,
+		}
+	};
+
+	/** Initializing IIO app init param. */
+	app_init_param.devices = iio_devices;
+	app_init_param.nb_devices = NO_OS_ARRAY_SIZE(iio_devices);
+	app_init_param.uart_init_params = max14906_uart_ip;
+
+	/** Initializing IIO app. */
+	ret = iio_app_init(&app, app_init_param);
+	if(ret)
+		goto app_error;
```

Comment:
this should go to error, because if init fails iio_app_remove does not need to be called, only if an error occurs after init is successful.